### ISSUE TITLE
Transactions from a specific address?

### ISSUE THREAD

**Issue by vindberg** [tezos-blue/client#2]
Hi, is it possible to use this library to get all the transactions (sent) for a specific address? I haven't been able to find it in the source.

Thanks for a well done lib.

**Comment by Hassmann** [comment]
Yes. If you want the full feature set and have the keys, the code you might be looking for is in Engine.cs:

`public async Task<Identity> ImportIdentity(string stereotype, string identityName, KeyPair keyPair)`

This will add the identity and all its contracts to your app engine. You will have access to the transaction history then via Account.Entries.

But if you just want to query any address not under your control, you can simply connect to the system and use IConnection.GetAccountEntries:

`Task<AccountEntry[]> GetAccountEntries(string accountID);`

Caveat: Currently this will only return the last 1000 transactions, pagination for earlier data is not implemented yet.

Feel free to contact me, I am happy to assist.

**Comment by vindberg** [comment]
Great to hear it. Im still trying to understand how you setup this. Is there an example of a real implementation of the Engine class? And where are all the RPC calls? HttpClient anywhere? :)

**Comment by Hassmann** [comment]
No RPC calls! You are connecting to an enriched layer 2 that communicates with a pool of managed nodes.
Can you give me a hint which OS you are targeting?
I will create a "Hello world" then.

**Comment by vindberg** [comment]
Ohh the backend system you mentions which is an implementation of the IConnection interface. Is that closed-source?

I want to use your library for an windows service process, keeping the blockchain in sync with a sql-database.

**Comment by Hassmann** [comment]
Oh, I see. That is exactly one of the things my backend does: To keep a geo-redundant cache of interpreted transactions in sync with the chain.
So people won't have to write services for SQL databases :)
If you don't want to build on this and do it yourself, I might isolate and publish my RPC interface code. I will have a look at the dependencies.

**Comment by Hassmann** [comment]
Or, possibly this might be a match for you?

https://github.com/mscappini/Tz.Net

Haven't checked it...

**Comment by vindberg** [comment]
Any code you are willing to share is much appreciated :)

You can also email me at av@validators dotcom if you want money for it. You using code-first?

**Comment by vindberg** [comment]
Any news on this @Hassmann?
